addons/mass_mailing/models/mailing.py

```diff
@@ -650,3 +650,26 @@ def _image_to_url(b64image: bytes):
             return lxml.html.tostring(root)
 
         return body_html
+
+    def _prepare_mail_body(self, body=None):
+        """Prepare the email body before sending.
+        Add the text preview at the beginning of the mail. The preview text is displayed
+        bellow the mail subject of most mail client (gmail, outlook...).
+        We add the character `&zwnj;` (zero-width non-joiner) to fill the end of the
+        preview in order to not have the beginning of the mail at the end of the preview
+        (don't work with simple space as the content is trimmed).
+        https://litmus.com/blog/the-ultimate-guide-to-preview-text-support
+        """
+        self.ensure_one()
+
+        body = body or self.body_html
+        preview = re.findall(r'\>([^<>]+)\<', body)
+        preview = ' '.join([p.strip() for p in preview if p.strip()])[:500]
+
+        html_preview = f"""
+            <div style="display:none;font-size:1px;height:0px;width:0px;opacity:0;">
+              {tools.html_escape(preview)} {'&zwnj;&nbsp;' * 500}
+            </div>
+        """ if preview else ''
+
+        return html_preview + body
```

Comment:
Meh, the + is quite harsh. We could try to be a bit more smart, like what we do with append_content_to_html .